services/node_create.proto

```diff
@@ -27,62 +25,111 @@ option java_multiple_files = true;
+ * - A `NodeCreateTransactionBody` MUST be signed by the governing council.
+ * - The newly created node information SHALL be added to the network address
+ *   book information in the network state.
+ * - The new entry SHALL be created in "state" but SHALL NOT participate in
+ *   network consensus and SHALL NOT be present in network "configuration"
+ *   until the next "upgrade" transaction (as noted below).
+ * - All new address book entries SHALL be added to the active network
+ *   configuration during the next `freeze` transaction with the field
+ *   `freeze_type` set to `PREPARE_UPGRADE`.
+ *
+ * ### Record Stream Effects
+ * Upon completion the newly assigned `node_id` SHALL be in the transaction
+ * receipt.
  */
 message NodeCreateTransactionBody {
-
     /**
-     * Node account id, mandatory field, ALIAS is not allowed, only ACCOUNT_NUM.
-     * If account_id does not exist, it will reject the transaction.
-     * Multiple nodes can have the same account_id.
+     * A Node account identifier.
+     * <p>
+     * This account identifier MUST be in the "account number" form.<br/>
+     * This account identifier MUST NOT use the alias field.<br/>
+     * If the identified account does not exist, this transaction SHALL fail.<br/>
+     * Multiple nodes MAY share the same node account.<br/>
+     * This field is REQUIRED.
      */
-    AccountID account_id = 1;
+    proto.AccountID account_id = 1;
 
     /**
-     * Description of the node with UTF-8 encoding up to 100 bytes, optional field.
+     * A short description of the node.
+     * <p>
+     * This value, if set, MUST NOT exceed 100 bytes when encoded as UTF-8.<br/>
+     * This field is OPTIONAL.
      */
     string description = 2;
 
     /**
-     * Ip address and port, mandatory field. Fully qualified domain name is
-     * not allowed here. Maximum number of these endpoints is 10.
-     * The first in the list is used as the Internal IP address in config.txt,
-     * the second in the list is used as the External IP address in config.txt,
-     * the rest of IP addresses are ignored for DAB phase 2.
+     * A list of service endpoints for gossip.
+     * <p>
+     * These endpoints SHALL represent the published endpoints to which other
+     * consensus nodes may _gossip_ transactions.<br/>
+     * This list MUST NOT be empty.<br/>
+     * This list MUST NOT contain more than `10` entries.<br/>
+     * The first two entries in this list SHALL be the endpoints published to
+     * all consensus nodes.<br/>
+     * All other entries SHALL be reserved for future use.
+     * <p>
+     * Each network may have additional requirements for these endpoints.
+     * A client MUST check network-specific documentation for those
+     * details.<br/>
+     * <blockquote>Example<blockquote>
+     * Hedera Mainnet _requires_ that address be specified, and does not
+     * permit DNS name (FQDN) to be specified.<br/>
+     * Mainnet also requires that the first entry be an "internal" IP
```

Comment:
I don't know if this impacts what is written here, in the code the effective difference between "internal" and "external" is the order in which the connection attempts are made.  If the "internal" address resolves, we attempt to connect to that address first.  Once we start having consensus rosters, the generalized treatment will be to attempt connection in order of the list and not make a categorical discrepancy between "internal" and "external".

Reply:
That will be, as I understand it, something visible in config.txt; and does not affect what is stored here in this phase.
Each `Node` from the in-state store will become an entry in `config.txt` as part of the freeze/upgrade process, and certificates (with their public keys) will be written to PEM files on disk.
As to later, the network address book (as distinct from the internal platform concept of the "roster") will need to track more than just consensus nodes, and will be used by more than just consensus nodes. Changes and/or adjustments needed for those later requirements are deferred to later consideration (c.f. Michael Heinrichs' comments above).